libs/linker/billsToOperation/operationsFilters.js

```diff
@@ -37,38 +38,71 @@ const isHealthBill = bill => {
 
 const filterByIdentifiers = identifiers => {
   identifiers = identifiers.map(i => i.toLowerCase())
-  return operation => {
+  const identifierFilter = operation => {
     const label = operation.label.toLowerCase()
     return some(identifiers, identifier => includes(label, identifier))
   }
+  return identifierFilter
 }
 
-const filterByDates = ({ minDate, maxDate }) => operation => {
-  return isWithinRange(operation.date, minDate, maxDate)
+const filterByDates = ({ minDate, maxDate }) => {
+  const dateFilter = operation => {
+    return isWithinRange(operation.date, minDate, maxDate)
+  }
+  return dateFilter
 }
 
-const filterByAmounts = ({ minAmount, maxAmount }) => operation => {
-  return operation.amount >= minAmount && operation.amount <= maxAmount
+const filterByAmounts = ({ minAmount, maxAmount }) => {
+  const amountFilter = operation => {
+    return operation.amount >= minAmount && operation.amount <= maxAmount
+  }
+  return amountFilter
 }
 
-const filterByCategory = bill => operation => {
-  return isHealthBill(bill)
-    ? isHealthOperation(operation) || isUncategorizedOperation(operation)
-    : !isHealthOperation(operation) || isUncategorizedOperation(operation)
+const filterByCategory = bill => {
+  const isHealth = isHealthBill(bill)
+  const categoryFilter = operation => {
+    return isHealth
+      ? isHealthOperation(operation) || isUncategorizedOperation(operation)
+      : !isHealthOperation(operation) || isUncategorizedOperation(operation)
+  }
+  return categoryFilter
+}
+
+/**
+ * Check that the sum of the reimbursements + the amount of the bill is not
+ * greater that the amount of the operation
+ */
+const filterByReimbursements = (bill, options={}) => {
+  const reimbursementFilter = operation => {
+    const sumReimbursements = sumBy(operation.reimbursements, 'amount')
+    return (sumReimbursements + bill.amount) <= -operation.amount
+  }
+  return reimbursementFilter
 }
 
 // combine filters
 
 const operationsFilters = (bill, operations, options) => {
   const filterByConditions = filters => op => {
-    return every(filters.map(f => f(op)))
+    for (let f of filters) {
```

Comment:
Isn't it the same result ?

Reply:
With this solution, as soon as there is a false, it returns false. I made this because during debugging, it was easier to see the reasons some bill was matched or not.